dashboard/test/models/course_test.rb

```diff
@@ -1,7 +1,25 @@
 require 'test_helper'
 
 class CourseTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test "should allow valid course names" do
+    create(:course, name: 'valid-name')
+  end
+
+  test "should not allow uppercase letters in course name" do
+    assert_raises ActiveRecord::RecordInvalid do
+      create(:course, name: 'UpperCase')
+    end
+  end
+
+  test "should not allow spaces in course name" do
+    assert_raises ActiveRecord::RecordInvalid do
+      create(:course, name: 'spaced out')
+    end
+  end
+
+  test "should allow uppercase letters if it is a plc course" do
```

Comment:
It's somewhat less important for these to be unified, given that PLC courses aren't being serialized to a file based on their name (unlike regular courses in the near future).